Frameworks/CoreGraphics/CGDataProvider.mm

```diff
@@ -66,7 +66,9 @@ - (NSUInteger)length {
 CGDataProviderRef CGDataProviderCreateWithURL(CFURLRef url) {
     NSString* path = [static_cast<NSURL*>(url) path];
     CGDataProvider* ret = [[CGDataProvider alloc] initWithContentsOfFile:path];
-    ret->filename = path;
+    if (ret) {
```

Comment:
this and CGDataProviderCreateWithFilename are pretty much the same logic.
re-use/refactor. rather than introducing the same code path.